netket/experimental/operator/_fermions_2nd.py

```diff
@@ -0,0 +1,769 @@
+            of_fermion_operator: openfermion.ops.FermionOperator object
+            n_orbitals: (optional) total number of orbitals in the system, default None means inferring it from the FermionOperator2nd. Argument is ignored when hilbert is given.
+            convert_spin_blocks: whether or not we need to convert the FermionOperator to our convention. Only works if hilbert is provided and if it has spin != 0
+
+        Returns:
+            A FermionOperator2nd object.
+        """
+        from openfermion.ops import FermionOperator
+
+        if hilbert is None:
+            raise ValueError(
+                "The first argument `from_openfermion` must either be an"
+                "openfermion operator or an Hilbert space, followed by"
+                "an openfermion operator"
+            )
+
+        if not isinstance(hilbert, AbstractHilbert):
+            # if first argument is not Hilbert, then shift all arguments by one
+            hilbert, of_fermion_operator = None, hilbert
+
+        if not isinstance(of_fermion_operator, FermionOperator):
+            raise NotImplementedError()
+
+        if convert_spin_blocks and hilbert is None:
+            raise ValueError("if convert_spin_blocks, the hilbert must be specified")
+
+        terms = list(of_fermion_operator.terms.keys())
+        weights = list(of_fermion_operator.terms.values())
+        terms, weights, constant = _collect_constants(terms, weights)
+
+        if hilbert is not None:
+            # no warning, just overwrite
+            n_orbitals = hilbert.n_orbitals
+
+            if convert_spin_blocks:
+                if not hasattr(hilbert, "spin") or hilbert.spin is None:
+                    raise ValueError(
+                        f"cannot convert spin blocks for hilbert space {hilbert} without spin"
+                    )
+                n_spin = hilbert._n_spin_states
+                terms = _convert_terms_to_spin_blocks(terms, n_orbitals, n_spin)
+        if n_orbitals is None:
+            # we always start counting from 0, so we only determine the maximum location
+            n_orbitals = _count_of_locations(of_fermion_operator)
+        if hilbert is None:
+            hilbert = SpinOrbitalFermions(n_orbitals)  # no spin splitup assumed
+
+        return FermionOperator2nd(hilbert, terms, weights=weights, constant=constant)
+
+    def __repr__(self):
+        return f"FermionOperator2nd(hilbert={self.hilbert}, n_operators={len(self._operators)})"
+
+    @property
+    def dtype(self) -> DType:
+        """The dtype of the operator's matrix elements ⟨σ|Ô|σ'⟩."""
+        return self._dtype
+
+    def copy(self, *, dtype: Optional[DType] = None):
+        if dtype is None:
+            dtype = self.dtype
+        op = FermionOperator2nd(
+            self.hilbert, [], [], constant=self._constant, dtype=dtype
+        )
+        op._operators = copy.deepcopy(self._operators)
+        op._constant = np.array(self._constant, dtype=dtype).item()
```

Comment:
this should be enough
```suggestion
        op = FermionOperator2nd(
            self.hilbert, [], [], constant=self._constant, dtype=dtype
        )
        op._operators = copy.deepcopy(self._operators)
```